lale/operators.py

```diff
@@ -497,6 +497,11 @@ def _final_estimator(self) -> Any:
                     model = impl._wrapped_model
                 elif isinstance(impl, sklearn.base.BaseEstimator):
                     model = impl
+                elif hasattr(impl, "estimator"):
```

Comment:
I am slightly concerned about how many things we are "magically" looking for here.  What is the motivation for specially supporting "estimator" and "mitigated"?  If it is only for a few impls, maybe we could find a better, more explicit design?  If this is a common convention for sklearn, it is probably ok, but we need to make sure it is documented somewhere.

Reply:
As described in my other comment, `classes_` is currently defined on `estimator` for post-estimator mitigators and directly on impl for in-estimator mitigators. These `elif` statements target one case each, but perhaps with some work we could group these cases into the `isinstance(impl, sklearn.base.BaseEstimator)` case?